### PR TITLE
[CLI-1180] prevent unnecessary modules rebuild

### PR DESCRIPTION
**Pull request by feons**
https://jira.appcelerator.org/browse/CLI-1180

### CODE REVIEW

File: lib/run.js [hunk 1]
```diff
@@ -44,10 +44,11 @@ function run(installBin, additionalArgs, cb, dontexit) {
...
 			var cmd = 'npm rebuild',
 				packageDir = path.join(installBin, '..', '..'),
+				versionsDir = path.join(packageDir, '..'),
 				isJSON = args.indexOf('json') === -1 ? false : true;
 
-			debug('run check package node version');
-			if (installBin && util.isNodeVersionChanged(installBin)) {
+			debug('run check package modules version');
+			if (installBin && util.isModuleVersionChanged(installBin)) {
 				util.outputInfo(chalk.yellow('\nYou are attempting to run appc ' +
```
Comment (ewanharris) [inline lib/run.js]: This log now returns something like the below first time round, could we possibly handle the migration for a user?
````
You are attempting to run appc 6.2.1 which was compiled for node undefined but you are now running node v6.10.3
Rebuilding package modules ...
````

**Comment by feons** [comment]
@ewanharris, nice catch! Updated.

**Comment by ewanharris** [comment]
@feons, so that got rid of the error message on first run, but if I change my node versions between this the `[Error: Module version mismatch.]` error reappears. It's edge casey for a normal user but given the frequency that we move between CLIs internally when testing it could cause problems

**Comment by feons** [comment]
@ewanharris, resolved conflicts. Thanks a mil!

**Pull request merged**